src/main.rs

```diff
@@ -0,0 +1,324 @@
+            sal_id: row[1].get_float().ok_or("Invalid Salary Id").expect("Parsing Error") as i32,
+            sal_date: (row[2].get_string().ok_or("Invalid Salaray Date").expect("Parsing Error") as &str).to_string(),
+            sal: row[3].get_float().ok_or("Invalid Salary").expect("Parsing Error") as f32,
+            status: (row[4].get_string().ok_or("Invalid Dept Strength").expect("Parsing Error") as &str).to_owned(),
+        };
+        salaries.push(salary);
+    }
+    Ok(salaries)
+}
+
+fn read_Leave_file(file_path: &str, sheet_name: &str) -> Result<Vec<LEAVE>, Box<dyn Error>> {
+
+    let mut workbook = open_workbook_auto(file_path).expect("Parsing Error");
+    let range = workbook.worksheet_range(sheet_name).expect("Parsing Error");
+
+    let mut leaves = Vec::new();
+
+    for row in range.rows().skip(1) {
+        //println!("{:?}",row);
+        let emp_id = row[0].get_float().ok_or("Invalid EMP ID").expect("Parsing Error") as i32;
+        //println!("{:?}",emp_id);
+        let leave_id = row[1].get_float().ok_or("Invalid Leave ID").expect("Parsing Error") as i32;
+
+        let leave_from_date_str = row[2].get_string().unwrap_or_default();
+        let leave_to_date_str = row[3].get_string().unwrap_or_default();
+
+        let leave_from = NaiveDate::parse_from_str(leave_from_date_str, "%e-%m-%Y")
+            .unwrap_or_else(|_| NaiveDate::from_ymd(1970, 1, 1)); // Use default date if parsing fails
+        let leave_to = NaiveDate::parse_from_str(leave_to_date_str, "%e-%m-%Y")
+            .unwrap_or_else(|_| NaiveDate::from_ymd(1970, 1, 1));
+
+        let leave_type = row[4].get_string().ok_or("Invalid Leave Type").expect("Parsing Error").to_string();
+
+        let leave_entry = LEAVE {
+            emp_id,
+            leave_id,
+            leave_from,
+            leave_to,
+            leave_type,
+        };
+
+        leaves.push(leave_entry);
+    }
+
+    Ok(leaves)
+}
+
+fn max_days_in_month(year: i32, month: u32) -> u32 {
+    let first_day_of_next_month = if month == 12 {
+        NaiveDate::from_ymd(year + 1, 1, 1)
+    } else {
+        NaiveDate::from_ymd(year, month + 1, 1)
+    };
+
+    let last_day_of_month = first_day_of_next_month.pred(); 
+    last_day_of_month.day() 
+}
+
+fn generate_output(
+    emp_data: Vec<EMPLOYEE>,
+    dept_data: Vec<DEPARTMENT>,
+    sal_data: Vec<SALARYENTRY>,
+    leave_data: Vec<LEAVE>,
+    output_path: &str,
+) -> Result<(), Box<dyn Error>> {
+   
+    let dept_map: std::collections::HashMap<i32, String> = dept_data
+        .into_iter()
+        .map(|dept| (dept.dept_id, dept.dept_title))
+        .collect();
+    
+    let current_month = Utc::now().month();
+
+    let file = File::create(output_path).expect("Parsing Error");
+    let mut write_handler = BufWriter::new(file);
+
+    writeln!(
+        write_handler,
+        "Emp ID~#~Emp Name~#~Dept Title~#~Mobile No~#~Email~#~Salary Status~#~On Leave"
+    ).expect("Parsing Error");
```

Comment:
Give the correct expect messages. It should be understood by the end-user or yourself when debugging.